man/man5/zfs-module-parameters.5

```diff
@@ -1229,6 +1251,91 @@ Rate limit delay zevents (which report slow I/Os) to this many per second.
 Default value: 20
 .RE
 
+.sp
+.ne 2
+.na
+\fBzfs_unflushed_max_mem_amt\fR (ulong)
+.ad
+.RS 12n
+Upper-bound limit for unflushed metadata changes to be held by the
+log spacemap in memory (in bytes).
+.sp
+Default value: \fB1,073,741,824\fR (1GB).
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_unflushed_max_mem_ppm\fR (ulong)
+.ad
+.RS 12n
+Percentage of the overall system memory that ZFS allows to be used
+for unflushed metadata changes by the log spacemap.
+(value is calculated over 1000000 for finer granularity).
+.sp
+Default value: \fB1000\fR (which is divided by 1000000, resulting in
+the limit to be \fB0.1\fR% of memory)
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_unflushed_log_block_max\fR (ulong)
+.ad
+.RS 12n
+Describes the maximum number of log spacemap blocks allowed for each pool.
+The default value of 262144 means that the space in all the log spacemaps
+can add up to no more than 262144 blocks (which means 32GB of logical
+space before compression and ditto blocks, assuming that blocksize is
+128k).
```

Comment:
Added. `mandoc` output:
```
       zfs_unflushed_log_block_max (ulong)
                   Describes the maximum number of log spacemap blocks allowed
                   for each pool.  The default value of 262144 means that the
                   space in all the log spacemaps can add up to no more than
                   262144 blocks (which means 32GB of logical space before
                   compression and ditto blocks, assuming that blocksize is
                   128k).

                   This tunable is important because it involves a trade-off
                   between import time after an unclean export and the
                   frequency of flushing metaslabs.  The higher this number
                   is, the more log blocks we allow when the pool is active
                   which means that we flush metaslabs less often and thus
                   decrease the number of I/Os for spacemap updates per TXG.
                   At the same time though, that means that in the event of an
                   unclean export, there will be more log spacemap blocks for
                   us to read, inducing overhead in the import time of the
                   pool.  The lower the number, the amount of flushing
                   increases destroying log blocks quicker as they become
                   obsolete faster, which leaves less blocks to be read during
                   import time after a crash.

                   Each log spacemap block existing during pool import leads
                   to approximately one extra logical I/O issued.  This is the
                   reason why this tunable is exposed in terms of blocks
                   rather than space used.

                   Default value: 262144 (256K).
```